jdaviz/core/marks.py

```diff
@@ -244,6 +244,11 @@ def set_x_unit(self, unit=None):
         super().set_x_unit(unit=unit)
         self._rest_value = (self._rest_value * prev_unit).to_value(unit, u.spectral())
 
+    def set_y_unit(self, unit=None):
+        # prevent spectral lines from entering unit conversion machinery so they maintain
+        # their position in the viewer at the rest + redshifted spectral value
```

Comment:
Is a regression test possible? I guess you can check marker values? 🤷‍♀️ 

Does this need a bug fix change log?

Reply:
For a regression test, the y-value range of the mark for the line will always be `[0, 1]` with this change so I don't think it is possible here. 

There was a mention of backporting in tagup for this PR, I think it would be for v4.0.x but wasn't certain. I will add it 4.0.1 change log now.

Reply:
do you think it would be possible for `PluginMark._on_global_display_unit_changed` to check for `LinearScale(0, 1)` and skip the call to `getattr(self, f'set_{axis}_unit')(msg.unit)` if so?  That way this can be applied in general across all marks (it looks now like the slice indicator handles this manually by overriding `_on_global_display_unit_changed` - maybe that could be removed as well then).

Reply:
Functionality was maintained with this change and the update to no longer override for the slice indicator also works!